### ISSUE TITLE
[Perf][Qwen3-Omni] Replace per-row CPU deque for talker future-text with a device-resident tensor + cursor

### ISSUE THREAD

**Issue by Hayden727** [sgl-project/sglang-omni#485]
## Problem

During talker decode, future-text embedding rows are injected one row per step and fused with the audio-feedback row before being fed into forward. Today these rows are kept as a Python `collections.deque` of **N independent 1-row CPU tensors**, which forces:

- producer side: one `.detach().cpu()` per row — N small D2H copies;
- consumer side: one `.to(device, dtype)` per decode step — N small H2D copies;
- a Python deque of N independent small objects in between — poor cache locality.

The source is a device-resident 2D tensor, the destination is the GPU, and the queue exists only to bridge the two. We disassemble it row by row, move each row to CPU, then move each row back to GPU. Audio decode often runs for thousands of steps, so the accumulated PCIe dispatch + Python-deque overhead is measurable. The producer-side `.cpu()` is also a **synchronous D2H** sitting on the thinker streaming critical path.

### Producer side: per-row `.cpu()`

`sglang_omni/models/qwen3_omni/components/talker_prefill.py:330-338`

```python
def tensor_rows_to_queue(
    self, tensor: torch.Tensor | None
) -> collections.deque[torch.Tensor]:
    queue: collections.deque[torch.Tensor] = collections.deque()
    if not isinstance(tensor, torch.Tensor) or tensor.numel() == 0:
        return queue
    for row in tensor:
        queue.append(row.detach().cpu())
    return queue
```

Later thinker-chunk / EOS appends (same file at `:259` and `:274`) also go row-by-row through `.cpu()` before being enqueued.

### Consumer side: per-decode-step `.to(device)`

`sglang_omni/models/qwen3_omni/talker_model_runner.py:267-275`

```python
combined = feedback.to(device=device, dtype=dtype).reshape(-1)
next_text = QwenTalkerModelRunner._peek_left(
    getattr(data, "pending_text_queue", None)
)
if next_text is not None:
    combined = combined + next_text.to(
        device=device, dtype=dtype,
    ).reshape(-1)
```

Every decode step triggers a `next_text.to(device, dtype)` — a single small H2D copy of one `[hidden_size]` row.

## Proposed direction

Change the backing form of `pending_text_queue` from "deque of N 1-row CPU tensors" to a **device-resident 2D tensor plus a cursor**:

- Producer keeps `future_text_rows` from `build_prefill_input(...)` as a device-resident `[N, hidden]` tensor;
- Incremental appends (`append_text_chunk` / `mark_thinker_done`) extend on-device, avoiding `.cpu()`;
- Consumer slices one row by cursor at decode time, eliminating per-step `.to(device, dtype)`.

`torch.cat` vs preallocated ring buffer is a benchmark call.

## Scope

- Only `pending_text_queue` is targeted; `pending_feedback_queue` is intentionally left alone.
- Currently only Qwen3-Omni talker uses this queue (Fish / Ming-Omni / Voxtral-TTS do not), so no framework-level abstraction yet.

## Relation to #484

Orthogonal: #484 optimizes the `.cpu().tolist()` at the moment of request construction; this issue optimizes the queue's backing form across the entire decode lifetime after the request exists. Same direction of benefit (TTFT + ITL), no conflict, expected to stack.

## Acceptance Metrics

- Talker per-decode-step wall-clock drops.
- Long-audio ITL P50 / P95 improves.
- WER / waveform parity does not regress.

**Comment by zhaochenyang20** [comment]
@JingwenGu0829

**Comment by yxs** [comment]
@JingwenGu0829 
Some validation data I ran  before.

nsys profile on talker (2× H200, c=16, n=200 EN VC=False stream, 30s steady-state):

| metric | value |
|---|---:|
| `cudaMemcpyAsync` total | 1.79 s / 30 s |
| `cudaMemcpyAsync` instances | 4,669 |
| `cudaStreamSynchronize` total | 316 ms |
| Total GPU kernel compute | ~18.6 ms |
| `cudaGraphLaunch_v10000` avg | 324 μs × 105 instances |

GPU compute is ~0.06% of the capture window — the H2D/D2H traffic this issue targets is most of where wall goes.

A 1R combined patch (this issue's direction + #484) on top of #487 didn't move bulk `latency_mean` (within ~1.5σ noise band), so the right validation surface is likely TTFT / ITL P95, not aggregate latency.
